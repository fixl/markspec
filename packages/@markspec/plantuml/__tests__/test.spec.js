const plantuml = require("..");

const diagram = `
@startuml
A-->B: request
B-->A: ack
@enduml`;

const erroneousDiagram = `@start`;

describe("plantuml", () => {
  it("renders a svg diagram", async () => {
    const uri = plantuml.toUriSync(diagram, 'svg');
    expect(uri).toMatchSnapshot();
  });
  it("renders a png diagram by defult", async () => {
    const uri = plantuml.toUriSync(diagram);
    expect(uri).toMatchSnapshot();
  });
});
