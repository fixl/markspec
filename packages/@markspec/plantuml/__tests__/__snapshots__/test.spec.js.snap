// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plantuml renders a png diagram by defult 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGwAAACgCAIAAABmLh5VAAAAKXRFWHRjb3B5bGVmdABHZW5lcmF0ZWQgYnkgaHR0cDovL3BsYW50dW1sLmNvbREwORwAAAEYaVRYdHBsYW50dW1sAAEAAAB4nE2OTWsCMRRF9w/8D2+pixmSqVqbRfGj0iIOSqPTZYlOKqFOMk1eLP77RkXo8l7OvZxxIOUpNscOTLLseSrQ65+oA3VgmvJEoNp/d2CsbX2FYH1UlrblEk/aB+Ms8rxg/Clng65UhJPWY8GQj0R/IB6GOJvLDV6AHnRf10sMLvq9xtoE8mYXKT30YKFOCt+jJdNogZfU3ZQ9lPN7iXN7Mt7ZRluCRVXeIHxzJFtHV3jYz6aGUGqfxLAqb6fVTVIky1HOPvljke04h1WrvSJjDyjPgXQj8MPY2v0G5AxW8t+M5Qxe9JeKR0oSe1enkcBZy4tBAUtlD1EdkrO2MHPJ1Z8FbiX8ATW1aS3tTbewAAAJWElEQVR42u2dDUxU2RXHp7U4odaPSaYspcZQ1ywhE2JsXKoxQSSmhFBKWWNCTBpT3Kwhu4Z1iYlYEzCAbewMpjWE7GIVBjM7O0pGhqWM0BGWNRNE6DRpDDphCxWB5SMqC6sDovTv3uY5nRlmZIZ333vjObkxZ573vMf53XPvuTdz4KkWSSIWFSEQC+L8w2/GO/pWpM0/mpHQPT6OBIYIG7MqdUXaRGefhBD5OBIM4p2z+vEOc9jtTpVeJhDFdiQYRNg/f/6PsNvX180ygSi2I6JDtFfVOp1Ol8vldrtHRkamp6cVCjGII6JDtJQZbDabw+Ho6enB4ycnJxUKMYgjokNsKDltMpmsVisej2HEGCoUYhBHRIdoPF5ZV1eHx2MYMR0GBwcVCjGII6JD/KxUj2fX19dbLBaMISaCQiEGcYQgEkSCSBAJYsQQHY6PVSpVW1uNciGqvCQ2Vp2ZudPtbuIKMS9vz7FjB3Nz0xUNUdA9npvl5e/rdG/ygzgy0rZx4xtQEhMTxsbaowAia4hHfhAxaCdOHIJy8uS7p04VRgHEhYXeysoP9PqP+EHctCl+YKAZyuBgC4IxCtZEJvX15Zwg2u3V6enbhY8ZGW8vlV6UFYkNDRVJSYmcIObkpPkMIK5Ex5qoVq/mAXF42I65jHHzHkNcCZhelAVxfv6WVruBB8TS0sP+q++ZMx8GTC8Kgvj4cTdSJTZtPCBu3foWnudzcXbWmZKyRdGJRaNZd+RIvr9rdOyjszNBJIgEkSASRIJIEAmiSBDvVOlhH3ZjdUBygCi2I6KX1skBotiOLFHk+WhmorOPNXtVraXM0FBy2ni8EjdadrtolBAiH0dClxs7nU6bzWYymeoiEKnKSPg4Ehqiy+UCeKvVCvv6sETCgiY+joSGiLjt6emBJUbAEpZIWFrHx5HQEIEcNmCPGHb4ycXDJY5QImGRJx9HQkNEb1CHGVYBt5/8Oebn7lACQ5jjJriVx+ORCqJ4jkT6eyzI/YtRIZE4QhAJYnRA/FfpJ9EBMRJH6Hf7VkAIIkEkiASREgttcWifSBAJIkGkxEInFtonEkQFydTUFEGMVOLi4ujEErE/KhV/R0Tc4sCf6urqzZs3Q3/w4EFWVlZMTExmZiZ01mFiYiItLU2r1RqNRsF5bwqCHtC8paUFV1atWpWSktLZ2cn6M4mefSKcMZvNT548gV5YWHj//n0oJpOpqKiIdSgoKACIZ8+eQQkOMaA5CII+lGvXrrGhiiQS5QsRgJiOcGMKmMbHxzNdo9GwDqOjo8EhBjTPzs4+cOAACD59+jTy6SxfiN66IJiA7KKgeHcOCDGgORLx3r17EY9r1651uVwKhhhkPfb2Z/369f4dEFMsEqenp/0hzs3NCXpAcya4w4ULFxISEhScWF4xUWJRu3fv3osBN5t37NjBLhYXFzc2NkI5evSo0BmR1draCjTeFwOaJycnY0mFgv6wEsxnZnj/sUFOEDH1kFjhITJpf38/uwhv8/Ly1Gq1wWAQOjMimKHnz58XLgY07+7u1ul0mN24zmhC9u3bhxsKz/12ePyr2qtPxqaUCpHP/i6kjLY6sd5d+dHu8Y4+ghi+DHxylVW7fr7lnf4/XfJMPozOE4v3HBRDhi61mr+/42pc5t93HrqyZrcz//c+gUnfsSyD44upvSatPfV3zYm/8Q5MifeJaGwY8a/M9bbtB71r2Zt+ksWUe5cd9PXAK8nX7TevrNuDGS1AtKh3df26eGZgeJG+Y3l1go3rM17g+94vmhNz3X/57PnTBbmfWOQj//m0zT/0lHFikYmwfaLlBzt9Qi8K94kiCU4syL+zg6OvxYlF0UIQo+jEIrnQiWUFhPaJBJEgEkRKLHRioX0iQeT7I6pUBJEgyiCx8IGoyBNLd3f3tm3bYmJi4uLiWF3S4nfVX7m5ubGxsUlJSU6n0weiwWAoKyujfeJLSU5Ovn79OpTGxkahSqSwsLC9vR1Kc3MzOnhDrKmpKS4ups126Amr1WqFQjLv/7106VJGRgadWHxlamqqoqIiPz8fESdA9K4T84a4f/9+dGMlinRieSmpqalY4JqamoaGhgSIGo1mqTi1WCwFBQV0Yvk/Qfa4e/fu3NxcUVGRd/XXjRs3oGC59FkTIUhEt2/fpn3iS0EMAhNSyrlz57xrs3NycpCydTpdb2+vD8Surq6srKzXEaJn8uFQw99WvIbodYE4+eU/WbnBaKuTzs7LW48RdP1/NDYlZLPvy/99sVkRIORyYkHodf2q2LJ612X1LlZuMNL8pVKiSeJ9ohB6lu/YsXb5h2kKIrgoh9I6/3brvT/Iv9JO0NlHiSPx7tlPP/9ZXuOGjMtrdv+vdGj1rvtXv6BIXPZ6PN7R1/3bMkxq649/qTiO8jqxCIHJQvKrvzYtRruIuE9EYNp+mk37xJUJTPc5y7fD4wSRhO+JRaFCVWHK32wTRIJIEKPvxEKJhYT2iQQxOiHOP/wGJ98VafOPZiR0j48jor/oa6KzT0KIfBwJ+sq5s/rxDnPYjb2pTQ4QxXZE9JcfygGi2I6IDtFeVSvh65X4OCI6REuZQcIXffFxRHSIDSWnJXzlHB9HRIdoPF4p4csP+ThCb9WlVxMTRIJIEAmiHCA6HB+rVKq2thrlQvT+O7WxserMzJ1udxNXiHl5e44dO5ibm65oiILu8dwsL39fp3uTH8SRkbaNG9+AkpiYMDbWHgUQWUM88oOIQTtx4hCUkyffPXWqMAogLiz0VlZ+oNd/xA/ipk3xAwPNUAYHWxCMUbAmMqmvL+cE0W6vTk/fLnzMyHh7qfSirEhsaKhISkrkBDEnJ81nAHElOtZEtXo1D4jDw3bMZYyb9xjiSsD0oiyI8/O3tNoNPCCWlh72X33PnPkwYHpREMTHj7uRKrFp4wFx69a38Dyfi7OzzpSULYpOLBrNuiNH8v1do2MfnZ0JIkEkiASRIBJEgkgQRYJ4p0oP+7AbqwOSA0SxHRG9tE4OEMV2ZIkiz0czE519rNmrai1lhoaS08bjlbjRsttFo4QQ+TgSutzY6XTabDaTyVQXgUhVRsLHkdAQXS4XwFutVtjXhyUSFjTxcSQ0RMRtT08PLDEClrBEwtI6Po6EhgjksAF7xLAjLJGwyJOPI6EhojeowwyrgDssgSHMcRPcyuPxSAVRPEfo91hWQAgiQZSH/BfhxDnV7ohslwAAAABJRU5ErkJggg=="`;

exports[`plantuml renders a svg diagram 1`] = `"data:image/svg+xml,%3c%3fxml version='1.0' encoding='UTF-8' standalone='no'%3f%3e%3csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' contentScriptType='application/ecmascript' contentStyleType='text/css' height='161px' preserveAspectRatio='none' style='width:109px%3bheight:161px%3b' version='1.1' viewBox='0 0 109 161' width='109px' zoomAndPan='magnify'%3e%3cdefs%3e%3cfilter height='300%25' id='fii913vv8i4cw' width='300%25' x='-1' y='-1'%3e%3cfeGaussianBlur result='blurOut' stdDeviation='2.0'/%3e%3cfeColorMatrix in='blurOut' result='blurOut2' type='matrix' values='0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 .4 0'/%3e%3cfeOffset dx='4.0' dy='4.0' in='blurOut2' result='blurOut3'/%3e%3cfeBlend in='SourceGraphic' in2='blurOut3' mode='normal'/%3e%3c/filter%3e%3c/defs%3e%3cg%3e%3cline style='stroke: %23A80036%3b stroke-width: 1.0%3b stroke-dasharray: 5.0%2c5.0%3b' x1='21' x2='21' y1='39.6094' y2='120.3125'/%3e%3cline style='stroke: %23A80036%3b stroke-width: 1.0%3b stroke-dasharray: 5.0%2c5.0%3b' x1='88' x2='88' y1='39.6094' y2='120.3125'/%3e%3crect fill='%23FEFECE' filter='url(%23fii913vv8i4cw)' height='31.6094' style='stroke: %23A80036%3b stroke-width: 1.5%3b' width='23' x='8' y='3'/%3e%3ctext fill='black' font-family='sans-serif' font-size='14' lengthAdjust='spacingAndGlyphs' textLength='9' x='15' y='24.5332'%3eA%3c/text%3e%3crect fill='%23FEFECE' filter='url(%23fii913vv8i4cw)' height='31.6094' style='stroke: %23A80036%3b stroke-width: 1.5%3b' width='23' x='8' y='119.3125'/%3e%3ctext fill='black' font-family='sans-serif' font-size='14' lengthAdjust='spacingAndGlyphs' textLength='9' x='15' y='140.8457'%3eA%3c/text%3e%3crect fill='%23FEFECE' filter='url(%23fii913vv8i4cw)' height='31.6094' style='stroke: %23A80036%3b stroke-width: 1.5%3b' width='23' x='75' y='3'/%3e%3ctext fill='black' font-family='sans-serif' font-size='14' lengthAdjust='spacingAndGlyphs' textLength='9' x='82' y='24.5332'%3eB%3c/text%3e%3crect fill='%23FEFECE' filter='url(%23fii913vv8i4cw)' height='31.6094' style='stroke: %23A80036%3b stroke-width: 1.5%3b' width='23' x='75' y='119.3125'/%3e%3ctext fill='black' font-family='sans-serif' font-size='14' lengthAdjust='spacingAndGlyphs' textLength='9' x='82' y='140.8457'%3eB%3c/text%3e%3cpolygon fill='%23A80036' points='76.5%2c67.9609%2c86.5%2c71.9609%2c76.5%2c75.9609%2c80.5%2c71.9609' style='stroke: %23A80036%3b stroke-width: 1.0%3b'/%3e%3cline style='stroke: %23A80036%3b stroke-width: 1.0%3b stroke-dasharray: 2.0%2c2.0%3b' x1='21.5' x2='82.5' y1='71.9609' y2='71.9609'/%3e%3ctext fill='black' font-family='sans-serif' font-size='13' lengthAdjust='spacingAndGlyphs' textLength='43' x='28.5' y='67.1045'%3erequest%3c/text%3e%3cpolygon fill='%23A80036' points='32.5%2c98.3125%2c22.5%2c102.3125%2c32.5%2c106.3125%2c28.5%2c102.3125' style='stroke: %23A80036%3b stroke-width: 1.0%3b'/%3e%3cline style='stroke: %23A80036%3b stroke-width: 1.0%3b stroke-dasharray: 2.0%2c2.0%3b' x1='26.5' x2='87.5' y1='102.3125' y2='102.3125'/%3e%3ctext fill='black' font-family='sans-serif' font-size='13' lengthAdjust='spacingAndGlyphs' textLength='21' x='38.5' y='97.4561'%3eack%3c/text%3e%3c/g%3e%3c/svg%3e"`;
